File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/navdrawer/NavDrawerFragment.java
package com.jdroid.android.sample.ui.navdrawer;

import android.os.Bundle;
import android.view.View;

import com.jdroid.android.activity.ActivityLauncher;
import com.jdroid.android.fragment.AbstractFragment;
import com.jdroid.android.sample.R;

public class NavDrawerFragment extends AbstractFragment {

	@Override
	public Integer getContentFragmentLayout() {
		return R.layout.navdrawer_fragment;
	}

	@Override
	public void onViewCreated(View view, Bundle savedInstanceState) {
		super.onViewCreated(view, savedInstanceState);

		findView(R.id.leftCustomNavDrawer).setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				ActivityLauncher.INSTANCE.startActivity(getActivity(), LeftCustomNavDrawerActivity.class);
			}
		});
		findView(R.id.rightCustomNavDrawer).setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				ActivityLauncher.INSTANCE.startActivity(getActivity(), RightCustomNavDrawerActivity.class);
			}
		});
		findView(R.id.userNavDrawer).setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				ActivityLauncher.INSTANCE.startActivity(getActivity(), UserNavDrawerActivity.class);
			}
		});
		findView(R.id.noNavDrawer).setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				ActivityLauncher.INSTANCE.startActivity(getActivity(), NoNavDrawerActivity.class);
			}
		});
	}
}

File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/navdrawer/NavDrawerFragment.kt
package com.jdroid.android.sample.ui.navdrawer

import android.os.Bundle
import android.view.View

import com.jdroid.android.activity.ActivityLauncher
import com.jdroid.android.fragment.AbstractFragment
import com.jdroid.android.sample.R

class NavDrawerFragment : AbstractFragment() {

    override fun getContentFragmentLayout(): Int? {
        return R.layout.navdrawer_fragment
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        findView<View>(R.id.leftCustomNavDrawer).setOnClickListener {
            ActivityLauncher.startActivity(
                activity,
                LeftCustomNavDrawerActivity::class.java
            )
        }
        findView<View>(R.id.rightCustomNavDrawer).setOnClickListener {
            ActivityLauncher.startActivity(
                activity,
                RightCustomNavDrawerActivity::class.java
            )
        }
        findView<View>(R.id.userNavDrawer).setOnClickListener { ActivityLauncher.startActivity(activity, UserNavDrawerActivity::class.java) }
        findView<View>(R.id.noNavDrawer).setOnClickListener { ActivityLauncher.startActivity(activity, NoNavDrawerActivity::class.java) }
    }
}
